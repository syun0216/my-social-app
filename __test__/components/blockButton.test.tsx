import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableOpacity, Text } from 'react-native';
import BlockButton from '../../components/blockButton';

const correctClickText = 'correct click';
const contentText = 'hello world';
const clickFunc = () => correctClickText;
const testRender = renderer.create(
  <BlockButton clickFunc={clickFunc}>
    <Text>{contentText}</Text>
  </BlockButton>
);
const testInstance = testRender.root;

test('renders correctly', () => {
  const tree = testRender.toJSON();
  expect(tree).toMatchSnapshot();
});

test('click correctly', () => {
  const btn = testInstance.findAllByType(TouchableOpacity);
  expect(btn[0].props.onPress()).toEqual(correctClickText);
});
