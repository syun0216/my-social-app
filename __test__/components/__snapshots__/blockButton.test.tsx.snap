// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "justifyContent": "flex-start",
      },
      Object {
        "height": 150,
        "paddingBottom": 0,
      },
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#D5EF7F",
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 150,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text>
      hello world
    </Text>
  </View>
</View>
`;
